tests/framework/Optimizers/GeneticAlgorithms/discrete/unconstrained/testGAMaxwoRepTwoPointsCrossover.xml

```diff
@@ -0,0 +1,172 @@
+  <TestInfo>
+    <name>framework/Optimizers/GA.MaxwoReplacemenTwoPointsCrossovert</name>
+    <author>MohammadAbdo</author>
+    <created>2020-05-16</created>
+    <classesTested>GeneticAlgorithm</classesTested>
+    <description>
+      This test assesses the Genetic algorithm using the weighted sum found in myLocalSum.py function.
+      The nominal dimensionality of the test problem is 3.
+      The objective variable is ans. The problem in unconstrained, it is a maximization problem, and the sampling is from discrete variables without replacement.
+      The cross over mechanism used is the twoPointsCrossover algorithm
+    </description>
+    <analytic>
+      This test uses myLocalSum's analytic objective function.
+    </analytic>
+  </TestInfo>
+
+  <RunInfo>
+    <WorkingDir>MaxwoReplacementTwoPointsCrossover</WorkingDir>
+    <Sequence>optimize, print</Sequence>
+  </RunInfo>
+
+  <Steps>
+    <MultiRun name="optimize" re-seeding="2286">
+      <Input class="DataObjects" type="PointSet">placeholder</Input>
+      <Model class="Models" type="ExternalModel">myLocalSum</Model>
+      <Optimizer class="Optimizers" type="GeneticAlgorithm">GAopt</Optimizer>
+      <SolutionExport class="DataObjects" type="PointSet">opt_export</SolutionExport>
+      <Output class="DataObjects" type="PointSet">optOut</Output>
+    </MultiRun>
+    <IOStep name="print">
+      <Input class="DataObjects" type="PointSet">opt_export</Input>
+      <Input class="DataObjects" type="PointSet">optOut</Input>
+      <Output class="OutStreams" type="Print">opt_export</Output>
+      <Output class="OutStreams" type="Print">optOut</Output>
+    </IOStep>
+  </Steps>
+
+  <Distributions>
+    <UniformDiscrete name='uniform_dist_Repl_1'>
+      <lowerBound>1</lowerBound>
+      <upperBound>6</upperBound>
+      <strategy>withReplacement</strategy>
+    </UniformDiscrete>
+
+    <UniformDiscrete name='uniform_dist_woRepl_1'>
+      <lowerBound>1</lowerBound>
+      <upperBound>6</upperBound>
+      <strategy>withoutReplacement</strategy>
+    </UniformDiscrete>
+  </Distributions>
+
+  <Optimizers>
+    <GeneticAlgorithm name="GAopt">
+      <samplerInit>
+        <limit>20</limit>
+        <initialSeed>42</initialSeed>
+        <writeSteps>every</writeSteps>
+        <type>max</type>
+      </samplerInit>
+
+      <GAparams>
+        <populationSize>20</populationSize>
+        <parentSelection>rouletteWheel</parentSelection>
+        <reproduction>
+          <crossover type="twoPointsCrossover">
+            <crossoverProb>0.8</crossoverProb>
+          </crossover>
+          <mutation type="swapMutator">
+            <mutationProb>0.9</mutationProb>
+          </mutation>
+        </reproduction>
+        <fitness type="invLinear">
+          <a>2.0</a>
+          <b>1.0</b>
+        </fitness>
+        <survivorSelection>fitnessBased</survivorSelection>
+      </GAparams>
+
+      <convergence>
+        <objective>-1</objective>
+      </convergence>
```

Comment:
I have checked several definitions in the user manual, for GA (see attached picture), it seems to me the value should be always positive, in addition, we should use InputData to check it. @Jimmy-INL @mandd  

![image](https://user-images.githubusercontent.com/7321071/145283883-2f635f44-6f9d-44b6-90a4-f288d5d48006.png)

Reply:
@wangcj05, that's what I am saying. The algorithm will converge when $Objective = \epsilon^{obj}$. If I don't want to use the objective convergence and want the algorithm to carry on till it reaches the iteration limit, I select  $\epsilon^{obj}$ to be something impossible. For this problem, the obj is always positive, so I picked -1 so that conv_obj is always False (0). If I couldn't explain myself, let's discuss this face to face. Thanks.

Reply:
As discussed, the convergence on objective is not consistent across different optimizations, and it can be confusing, a new issue has been opened #1729